### DISCUSSION TITLE
What's the correct workflow for merging two subfolders of originals without losing PP metadata?

### DISCUSSION THREAD

**Discussion by aaronrudkin** [photoprism/photoprism#3433] [Q&A] [answered]
I have two subfolders inside my originals, let's say `A/` and `B/`, which represent the same event -- my wife's photos and my photos of the same event. I went into each photo and did appropriate edits. Labeling, face tagging, rotation of a few photos whose orientation setting was wrong, archiving some low quality or nearly duplicate photos etc. Now, later, I have decided I don't want to keep the subfolders separate on the original storage medium.

I know if I move all the files from `B/` to `A/` and reindex that the photos will transfer over no problem, but what I'm interested in is whether or not the reindexing process will "remember" all of the metadata edits made, or if I'll lose them. Also both folders are very large and I'd like to avoid having being weighed down by regenerating all the thumbnails etc. If this does happen to be a lossy operation, is there a preferred workflow to avoid this?

I suppose I could experiment but this seems like a case where respondents might actually be more well equipped to know everything that goes on downstream when you move a file and reindex, so I figured I'd ask.

(I understand that re-architecting the software to reflect this use case is not worth anyone's time or resources; just asking the best way to operate within the software as it exists today. And no rush, this is not urgent).

**Comment by lastzero** [comment, marked as answer]
Note that the metadata is not associated with a specific folder, but with the photo, which in turn refers to a filename prefix and one or more file hashes. So as long as the filename and file content (SHA1 hash) do not change at the same time, none of your edits should be lost.

If this is extremely important to you and the folders are very large, I would still ask you to try it first.

**Reply by aaronrudkin** [reply]
Oh, great. Architecturally that makes perfect sense. Thanks again for your time.

Can I beg your indulgence just a little more and ask what "filename prefix" means in this case -- everything before the extension? By this I mean if I were to rename a file that has a character that's legal in some filesystems and not in others (I have a surprising number of these), would that trigger the file being "forgotten"? 

Or do you just mean filename prefix in terms of option for auto-stacking files with the name schema "DSC_101.jpg" ,"DSC_101 (1).jpg", "DSC_101 (2).jpg", etc?

**Reply by lastzero** [reply]
The filename prefix includes the path and filename excluding the extension. As you noted, this is to stack related files.
